ports/geckolib/properties.mako.rs

```diff
@@ -184,6 +184,79 @@ def set_gecko_property(ffi_name, expr):
     }
 </%def>
 
+<%def name="convert_rgba_to_nscolor(rgba)">
+    {
+        let rgba = ${rgba};
+        (((rgba.alpha * 255.0).round() as u32) << 24) |
+        (((rgba.blue  * 255.0).round() as u32) << 16) |
+        (((rgba.green * 255.0).round() as u32) << 8) |
+         ((rgba.red   * 255.0).round() as u32)
+    }
+</%def>
+
+<%def name="convert_nscolor_to_rgba(color)">
+    {
+        use cssparser::RGBA;
+        let color = ${color};
+        RGBA {
+            red:    ((color        & 0xff) as f32) / 255.0,
+            green: (((color >>  8) & 0xff) as f32) / 255.0,
+            blue:  (((color >> 16) & 0xff) as f32) / 255.0,
+            alpha: (((color >> 24) & 0xff) as f32) / 255.0,
+        }
+    }
+</%def>
+
+<%def name="impl_color_setter(ident, gecko_ffi_name, gecko_currentcolor_mask_ffi_name=None)">
+    fn set_${ident}(&mut self, v: longhands::${ident}::computed_value::T) {
+        use cssparser::Color;
+        use gecko_style_structs as gss;
+        % if gecko_currentcolor_mask_ffi_name:
+        self.gecko.${gecko_currentcolor_mask_ffi_name} &= !(gss::BORDER_COLOR_SPECIAL as u8);
+        % endif
+        let result = match v {
+            Color::CurrentColor => {
+                % if gecko_currentcolor_mask_ffi_name:
+                self.gecko.${gecko_currentcolor_mask_ffi_name} |= gss::BORDER_COLOR_FOREGROUND as u8;
+                % else:
+                // FIXME(heycam): To handle this, we would need to get the
+                // current value of the color property.
+                println!("can't handle currentColor for ${ident}");
```

Comment:
And this could be set_current_color_flag, with an unimplemented!() instead of the FIXME + println.